pkgs/applications/virtualization/virtualbox/default.nix

```
@@ -85,7 +85,13 @@ in stdenv.mkDerivation {
   patches =
      optional enableHardening ./hardened.patch
      # Since VirtualBox 7.0.8, VBoxSDL requires SDL2, but the build framework uses SDL1
-  ++ optional (!headless) ./fix-sdl.patch
+  ++ optional (!headless) [ ./fix-sdl.patch
```

```suggestion
  ++ optionals (!headless) [ ./fix-sdl.patch
```

You should use `optionals` here, since you're wrapping twice the value using `optional`.